tests/facility/settings/chargeItemDefinitions.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { faker } from "@faker-js/faker";
+import { expect, test } from "@playwright/test";
+import { getFacilityId } from "tests/support/facilityId";
+
+test.use({ storageState: "tests/.auth/user.json" });
+
+test.describe(() => {
+  let facilityId: string;
+  let testData: {
+    title: string;
+    slug: string;
+    basePrice: string;
+    mrp: string;
+    purchasePrice: string;
+    description: string;
+    purpose: string;
+    url: string;
+  };
+
+  test.beforeEach(async ({ page }) => {
+    facilityId = getFacilityId();
+    const chargeItemName = faker.commerce.productName();
+    testData = {
+      title: chargeItemName,
+      slug: chargeItemName.replace(/\s+/g, "-").slice(0, 25),
+      basePrice: faker.commerce.price(),
+      mrp: faker.commerce.price(),
+      purchasePrice: faker.commerce.price(),
+      description: faker.commerce.productDescription(),
+      purpose: faker.commerce.productAdjective(),
+      url: faker.internet.url(),
+    };
+
+    await page.goto(
+      `/facility/${facilityId}/settings/charge_item_definitions/categories/f-${facilityId}-medications-charge_item_definition`,
+    );
+  });
+
+  test("validate required fields", async ({ page }) => {
+    await page.getByRole("button", { name: /add definition/i }).click();
+    await page.getByRole("button", { name: /create/i }).click();
+
+    // Title required
+    await expect(page.getByText(/title.*required/i)).toBeVisible();
+    // Slug required/length
+    await expect(page.getByText(/slug.*atleast 5.*atmost 25/i)).toBeVisible();
+    // Base Price required/invalid
+    await expect(page.getByText(/base price.*greater than 0/i)).toBeVisible();
+  });
+
+  test("create charge item definition", async ({ page }) => {
```

Comment:
1. add a test with mandatory fields only
2. verify the slug is auto generated
3. verify the slug 5-25 limit